### DISCUSSION TITLE
Type inference from L-values and return types

### DISCUSSION THREAD

**Discussion by ekolis** [dotnet/csharplang#4527] [Language Ideas] [answered]
If I have this code:

```csharp
IEnumerable<int> nums = Enumerable.Empty();
```

I get an error saying that the type parameter of `Empty` can't be inferred. But why can't it? I'm assigning to an `IEnumerable<int>`; couldn't the type parameter thus be inferred to be `int`?

The same goes for return types:

```csharp
public static IEnumerable<int> CreateEmptyList() => Enumerable.Empty();
```

**Comment by 333fred** [comment, marked as answer]
See https://github.com/dotnet/csharplang/discussions/92.
